src/main/java/wolox/training/services/OpenLibraryService.java

```diff
@@ -0,0 +1,58 @@
+package wolox.training.services;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Optional;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+@Service
+public class OpenLibraryService {
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    public Optional<Book> bookInfo(String isbn) throws Exception {
+        try {
+            String json = getJSON("https://openlibrary.org/api/books?bibkeys=ISBN:"+ isbn +"&format=json&jscmd=data");
+            JSONObject obj = new JSONObject(json);
+            obj = obj.getJSONObject("ISBN:"+isbn);
+            Book book = new Book();
+            book.setIsbn(isbn);
+            book.setTitle(obj.getString("title"));
```

Comment:
Maybe you could use static constants for these tags (subtitle, publishers, name, etc.) since you might use them multiple times.